src/common/ty/ty_printer.ml

```diff
@@ -223,12 +223,19 @@ let type_ ?(size=5000) ?(with_comments=true) t =
       | Get t -> fuse [
           Atom "get"; space;
           to_key key;
+          Atom "(";
+          Atom ")";
+          Atom ":";
           type_ ~depth t;
         ]
 
       | Set t -> fuse [
           Atom "set"; space;
           to_key key;
+          Atom "(";
+          type_ ~depth t;
```

Comment:
printing `t` twice seems odd to me. should the return type be void instead?

Reply:
```
const m = {set foo(value) {}}
let d = m.foo = 1
d // 1
```

Reply:
Also, I think this should be properties with variance annotations instead in types, but not sure if this is intended

Reply:
You also can declare any type in return, but it doesn't have effect, so perhaps setters should error on specifying return type entirely
```js
const bar = {
  set a(value: string): number {
    return 1;
  }
};
let d = (bar.a = '')
d // ''
```

Reply:
in `let d = m.foo = 1`, `d` is `1` because an expression statement evaluates to the right-hand side, not because the return type of the setter.

Reply:
@samwgoldman I think it still makes sense because you can't get `void` from setter

Reply:
But maybe setters and getters should be displayed as covariant and contravariant properties, and regular property if it is both

Reply:
IMO it's fine to just print `void` for the return type. We can either (1) ban return type annotations / enforce void return or (2) allow the annotations and just store the return type along with the setter property, even though it is ignored.